docs/contributing/999-rfcs/2024-01-15-console-sign-in.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "Wing Console Sign In"
+description: Requirements for Wing Console Sign In process
+---
+
+# Wing Console Sign In
+
+- **Author(s)**: @ainvoner
+- **Submission Date**: 2024-01-15
+- **Stage**: Proposal
+- **Stage Date**: 2024-01-15
+
+This document describes the Wing Console Sign In process and rationale.
+
+## Background
+
+As we seek to find **Sales and Marketing Opportunities**, it has become increasingly important to establish a more direct and meaningful connection with our users.
+One of the ways to achieve this is by implementing a sign-in flow within the Wing Console application.
+
+## Requirements
+
+New users of Wing Console will be required to sign in with their GitHub account before having access to the Wing Console application.
+
+Another requirement is to have only one user entity in Wing. This means that a user in Wing Console is also a user in Wing Cloud and vice versa.
+
+## User Experience
+
+### New Users
+
+New users that will `wing it` will see a sign in dialog blocking the Wing Console UI. The rationale behind this decision is that we want to show the user a teaser of Wing Console while asking them to sign in. 
+
+Once users will be signed in with their GitHub account they will be able to use Wing Console.
+
+### Singed In Users
+
+Signed in users to Wing Console will not need to sign in again. They will be able to use Wing Console as usual.
+
+### Offline
+
+If the user is offline (no internet connection) they will be able to use Wing Console as usual without the need to sign in.
+
+### Support of multiple users on the same machine
+
+There would be no support for multiple users on the same machine. 
+
+## User identity for Analytics  
+
+Events generated by signed in users will include the user's GitHub account email as an integral part of their user identity.
+In contrast, Wing CLI users are initially anonymous, and all analytics events originating from the CLI are sent anonymously with a unique identifier.
+However, when a user signs in to Wing Console, the unique identifier associated with CLI analytics events will be associated with the signed in user's identity, allowing us to link CLI analytics data to specific user accounts
+
+## Technical details
+
+Since we would like to have a single user entity in Wing, the process of signing in to Wing Console will use the same authentication mechanism as Wing Cloud.
+In fact, the user will actually be singing in to Wing Cloud and be redirected back to Wing Console without noticing.
+
+In order to make sure a sign in user won't need to sign in again, we will save the uses' authentication token in user's machine.
```

Comment:
```suggestion
In order to make sure a sign in user won't need to sign in again, we will save the uses' authentication token in user's machine.

## Next Steps

Add value for the user in exchange for signin up. Things which could be considered:

- Deploy to wing.cloud directly from the console. This could upload the working directory to a Bucket from which wing.cloud could deploy. The deployment status and the url would be shown in the local console.
- Offer a solution similar to ngrok, to make the locally running instance available for remote services (e.g. webhooks) or users (e.g. demo purposes)
```